pkg/action/git-credential.go

```diff
@@ -0,0 +1,189 @@
+package action
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/justwatchcom/gopass/pkg/store/sub"
+
+	"github.com/justwatchcom/gopass/pkg/store/secret"
+
+	"github.com/justwatchcom/gopass/pkg/fsutil"
+
+	"github.com/justwatchcom/gopass/pkg/out"
+
+	"github.com/justwatchcom/gopass/pkg/ctxutil"
+
+	"github.com/urfave/cli"
+)
+
+type gitCredentials struct {
+	Protocol string
+	Host     string
+	Path     string
+	Username string
+	Password string
+}
+
+func (c *gitCredentials) WriteTo(w io.Writer) (int64, error) {
+	var n int64
+	if c.Protocol != "" {
+		i, err := io.WriteString(w, "protocol="+c.Protocol+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Host != "" {
+		i, err := io.WriteString(w, "host="+c.Host+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Path != "" {
+		i, err := io.WriteString(w, "path="+c.Path+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Username != "" {
+		i, err := io.WriteString(w, "username="+c.Username+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Password != "" {
+		i, err := io.WriteString(w, "password="+c.Password+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	return n, nil
+}
+
+func parseGitCredentials(r io.Reader) (*gitCredentials, error) {
+	rd := bufio.NewReader(r)
+	c := &gitCredentials{}
+	for {
+		key, err := rd.ReadString('=')
```

Comment:
Why not just do
```
line, err := rd.ReadString('\n')
kv := strings.SplitN(line, "=", 2)
key = strings.TrimSpace(kv[0])
value = strings.TrimSpace(kv[1])
```
I admin that it's not much shorter, but that's the way I usually do this.

Reply:
It just seemed more straightforward to read the format exactly as it is described in the man page